lua/frecency/config.lua

```diff
@@ -167,40 +173,52 @@ end
 ---@return nil
 Config.setup = function(ext_config)
   local opts = vim.tbl_extend("force", Config.default_values, ext_config or {})
-  vim.validate {
-    recency_values = { opts.recency_values, "t" },
-    auto_validate = { opts.auto_validate, "b" },
-    bootstrap = { opts.bootstrap, "b" },
-    db_root = { opts.db_root, "s" },
-    db_safe_mode = { opts.db_safe_mode, "b" },
-    db_validate_threshold = { opts.db_validate_threshold, "n" },
-    debug = { opts.debug, "b" },
-    default_workspace = { opts.default_workspace, "s", true },
-    disable_devicons = { opts.disable_devicons, "b" },
-    enable_prompt_mappings={opts.enable_prompt_mappings,'b'},
-    filter_delimiter = { opts.filter_delimiter, "s" },
-    hide_current_buffer = { opts.hide_current_buffer, "b" },
-    ignore_patterns = { opts.ignore_patterns, "t" },
-    matcher = {
-      opts.matcher,
-      function(v)
-        return type(v) == "string" and (v == "default" or v == "fuzzy")
-      end,
-      '"default" or "fuzzy"',
-    },
-    max_timestamps = {
-      opts.max_timestamps,
-      function(v)
-        return type(v) == "number" and v > 0
-      end,
-      "positive number",
-    },
-    show_filter_column = { opts.show_filter_column, { "b", "t" }, true },
-    show_scores = { opts.show_scores, "b" },
-    show_unindexed = { opts.show_unindexed, "b" },
-    workspace_scan_cmd = { opts.workspace_scan_cmd, { "s", "t" }, true },
-    workspaces = { opts.workspaces, "t" },
-  }
+  if vim.version().minor >= 11 then
```

Comment:
I think `vim.fn.has('nvim-0.11') == 1` is more idiomatic (and probably faster since there's no need for parsing and table construction).